bloomrun.js

```diff
@@ -16,6 +16,7 @@ function BloomRun (opts) {
   this._isDeep = opts && opts.indexing === 'depth'
   this._buckets = []
   this._properties = new Set()
+  this._defaultResult = null
```

Comment:
### catchall 
IMHO either it is special, and you implement it outside of bloomrun, or it is "business as usual", and it follows the same 
rules of everything else from an API point of view. Having something _special_ that you add via normal APIs might trigger unsuspected bugs later on.

Neither this PR nor the current master follows this rule, so we need to fix.

### matchall

For me this is: `{ cmd: /.*/ }`. Is this what you mean?

From a code point of view, this pattern cannot be indexed by the bloom filters. This means we have `{}` to be indexed (and this is why these were not working before). Given _any_ object, if the bloom filters are not matching, then you are in the catchall space. _Before_ applying a catchall, we need to check the regexps.